zerver/models.py

```diff
@@ -1679,3 +1685,27 @@ class CustomProfileFieldValue(models.Model):
 
     class Meta(object):
         unique_together = ('user_profile', 'field')
+
+class Service(models.Model):
+    name = models.CharField(max_length=UserProfile.MAX_NAME_LENGTH, unique=True) # type: Text
```

Comment:
I don't think we want this to be tagged as globally unique; at best we'd want (realm, name) to be a unique pair.